Mathlib/Tactic/Measurability.lean

```diff
@@ -0,0 +1,40 @@
+/-
+Copyright (c) 2023 Miyahara Kō. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Miyahara Kō
+-/
+
+import Mathlib.Tactic.Measurability.Init
+import Mathlib.Algebra.Group.Defs
+
+/-!
+# Measurability
+
+We define the `measurability` tactic using `aesop`. -/
+
+attribute [aesop (rule_sets [Measurable]) unfold norm] Function.comp
+attribute [aesop (rule_sets [Measurable]) unfold norm] npowRec
+
+/--
+The `measurability` attribute used to tag continuity statements for the `measurability` tactic. -/
+macro "measurability" : attr =>
+  `(attr|aesop safe apply (rule_sets [$(Lean.mkIdent `Measurable):ident]))
```

Comment:
Is there a reason why the identifier is being constructed using an antiquotation and `Lean.mkIdent` rather than writing the identifier directly?
```suggestion
  `(attr|aesop safe apply (rule_sets [Measurable]))
```
(I don't know much about aesop yet, so this is me asking you to do the work to explain it to me rather than me investigating it myself.)

Reply:
Without this antiquotation, Lean inserts a macro scope and emits error.
```lean
@[simp, measurability]
theorem MeasurableSet.empty [MeasurableSpace α] : MeasurableSet (∅ : Set α) :=
  MeasurableSpace.measurableSet_empty _
#align measurable_set.empty MeasurableSet.empty
/-
no such rule set: 'Measurable._@.Mathlib.MeasureTheory.MeasurableSpaceDef._hyg.429'
  (Use 'declare_aesop_rule_set' to declare rule sets.
   Declared rule sets are not visible in the current file; they only become visible once you import the declaring file.)
-/
```

Reply:
@JLimperg I'm wondering if it should be `aesop`'s responsibility to resolve ruleset names when elaborating `rule_sets` syntax? I see that [on this line](https://github.com/JLimperg/aesop/blob/master/Aesop/Frontend/RuleExpr.lean#L459) it's doing `getId` directly, which gives the name including macro scopes. Should it clear macro scopes (since presumably these names are only ever global names) and then validate the name with perhaps `getRuleSetExtension`?

Mathlib is working around this in the [`continuity` tactic](https://github.com/leanprover-community/mathlib4/blob/master/Mathlib/Tactic/Continuity.lean) and in this PR by using `Lean.mkIdent`, which creates identifiers without macro scopes.

Reply:
I didn't do this because it makes the API less powerful: macros which hygienically create rule sets become impossible. But in practice, this is probably too much of an edge case to be relevant, so I've changed Aesop to remove the macro scopes.